core/src/main/java/org/bitcoinj/core/Wallet.java

```diff
@@ -2146,12 +2250,42 @@ public boolean maybeCommitTx(Transaction tx) throws VerificationException {
             // move any transactions that are now fully spent to the spent map so we can skip them when creating future
             // spends.
             updateForSpends(tx, false);
-            // Add to the pending pool. It'll be moved out once we receive this transaction on the best chain.
-            // This also registers txConfidenceListener so wallet listeners get informed.
-            log.info("->pending: {}", tx.getHashAsString());
-            tx.getConfidence().setConfidenceType(ConfidenceType.PENDING);
-            confidenceChanged.put(tx, TransactionConfidence.Listener.ChangeReason.TYPE);
-            addWalletTransaction(Pool.PENDING, tx);
+            
+            Set<Transaction> doubleSpendPendingTxns = findDoubleSpendsAgainst(tx, pending);
+            Set<Transaction> doubleSpendUnspentTxns = findDoubleSpendsAgainst(tx, unspent);
+            Set<Transaction> doubleSpendSpentTxns = findDoubleSpendsAgainst(tx, spent);
+            
+            if (!doubleSpendUnspentTxns.isEmpty() || 
```

Comment:
I am not following you. I am trying to cover the case of a peer sending us a new tx in conflict with a tx already building (I don't know why a peer would do that, but just in case).